framework/db/ActiveRecord.php

```diff
@@ -314,7 +332,13 @@ public static function getTableSchema()
      */
     public static function primaryKey()
     {
-        return static::getTableSchema()->primaryKey;
+        static $primaryKey = null;
+
+        if ($primaryKey === null) {
+            $primaryKey = static::getTableSchema()->primaryKey;
+        }
+
+        return $primaryKey;
```

Comment:
Unlikely to make sense.
Here you avoiding just single call of getTableSchema by the cost of extra memory usage, which stores exact(!) copy of TableSchema::primaryKey.